test/e2e/rollout_test.go

```diff
@@ -542,6 +548,231 @@ var _ = Describe("placing wrapped resources using a CRP", Ordered, func() {
+				},
+				{
+					Group:     testv1alpha1.GroupVersion.Group,
+					Kind:      testCustomResource.Kind,
+					Name:      testCustomResource.Name,
+					Version:   testv1alpha1.GroupVersion.Version,
+					Namespace: workNamespace.Name,
+				},
+				{
+					Group:   utils.CRDMetaGVK.Group,
+					Kind:    utils.CRDMetaGVK.Kind,
+					Name:    testResourceCRDName,
+					Version: utils.CRDMetaGVK.Version,
+				},
+			}
+		})
+
+		It("create the resources", func() {
+			Expect(hubClient.Create(ctx, &workNamespace)).To(Succeed(), "Failed to create namespace %s", workNamespace.Name)
+			Expect(hubClient.Create(ctx, &testCustomResource)).To(Succeed(), "Failed to create test custom resource %s", testCustomResource.GetName())
+		})
+
+		It("create the CRP that select the namespace and CRD", func() {
+			crp = buildCRPForSafeRollout()
+			crdClusterResourceSelector := placementv1beta1.ClusterResourceSelector{
+				Group:   utils.CRDMetaGVK.Group,
+				Kind:    utils.CRDMetaGVK.Kind,
+				Version: utils.CRDMetaGVK.Version,
+				Name:    testResourceCRDName,
+			}
+			crp.Spec.ResourceSelectors = append(crp.Spec.ResourceSelectors, crdClusterResourceSelector)
+			crp.Spec.Policy = &placementv1beta1.PlacementPolicy{
+				PlacementType: placementv1beta1.PickFixedPlacementType,
+				ClusterNames: []string{
+					memberCluster1EastProdName,
+				},
+			}
+			crp.Spec.Strategy.RollingUpdate.UnavailablePeriodSeconds = ptr.To(60)
+			Expect(hubClient.Create(ctx, crp)).To(Succeed(), "Failed to create CRP")
+		})
+
+		It("should update CRP status as expected", func() {
+			crpStatusUpdatedActual := customizedCRPStatusUpdatedActual(crpName, wantSelectedResources, []string{memberCluster1EastProdName}, nil, "0", false)
+			Eventually(crpStatusUpdatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update CRP status as expected")
+		})
+
+		It("should place the resources on member cluster", func() {
+			workResourcesPlacedActual := waitForTestResourceToBePlaced(memberCluster1EastProd, &testCustomResource)
+			Eventually(workResourcesPlacedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to place work resources on member cluster %s", memberCluster1EastProd.ClusterName)
+		})
+
+		It("should update custom resource", func() {
+			Eventually(func() error {
+				var cr testv1alpha1.TestResource
+				err := hubClient.Get(ctx, types.NamespacedName{Name: testCustomResource.Name, Namespace: workNamespace.Name}, &cr)
+				if err != nil {
+					return err
+				}
+				cr.Spec.Foo = "bar1" // Previously was "foo1"
+				return hubClient.Update(ctx, &cr)
+			}, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update custom resource")
+		})
+
+		It("should not update the resource on member cluster", func() {
+			// subtracting a second because transition between IT takes ~1 second
+			unavailablePeriod := time.Duration(*crp.Spec.Strategy.RollingUpdate.UnavailablePeriodSeconds)*time.Second - time.Second
```

Comment:
better to leave a bit more buffer so the test case is not flaky, how about 5 seconds.